kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -249,43 +266,56 @@ private boolean checkRotationOrAppend(
       String encodedPartition,
       long now
   ) {
+    // rotateOnTime check must go before writeRecord.
```

Comment:
```suggestion
    // rotateOnTime is safe to go before writeRecord, because it is acceptable even for a faulty record to trigger time-based rotation if it applies 
```